src/multi-select.js

```diff
@@ -94,6 +96,40 @@ class MultiSelect extends PureComponent {
 				highlightedIndex: 0
 			});
 		}
+
+		if (!isEqual(prevProps.defaultSelected, this.props.defaultSelected)) {
+			this.setState({ // eslint-disable-line react/no-did-update-set-state
+				selected: prevProps.defaultSelected
+			});
+		}
```

Comment:
This worries me a little. `defaultSelected` sounds like it applies those values at the start (`default`), but this means if those change, it'll reset the current selected state. 

What is the use-case for this? Is it for when the items needs to change without unmounting? 

If it's to control the current selections, we should probably turn it into a controlled component. Or have the option at least, with a warning if it starts one way and changes in the middle, same as React on the web and inputs.

Reply:
That's spot on; the intended use case is for updating selected items on the fly - without unmounting. This feels very close to a controlled component already, what did you have in mind? Updating defaultSelected to selected would tighten the expectations around its use.